GH3207 - translations updates (#3487)
magento/devdocs: guides/v2.1/frontend-dev-guide/translations/xlate.md

  - Frontend
---

The Magento application enables you to localize your store for multiple regions and markets. We improved the localization and customization of Magento instances by making translation dictionaries easier to update and by maintaining a reduced amount of code coupling and duplication.

> We also accept [Community Engineering contributions](#translations-project) using CrowdIn for translations. We plan to provide translation packs through the project.

Reviewer: Also, we...

"Address %1 of %2","Adresse %1 von %2"
```

> *UI text strings* are the text elements in the Magento Admin, including field names, instructions, notifications, and table and grid labels. When localizing Magento, you use translations of these strings using the translation dictionary content.

Author: Yep, that needed a revision. thanks!

A *language package* is basically a collection of translation dictionaries for a particular language together with meta-information. You can also distribute language packages to other merchants if you wish. [More information about language packages](#m2devgde-xlate-languagepack).

{:.bs-callout .bs-callout-info}
To create a language package, the `.csv` file requires additional columns that specify the themes or modules in which the translations were found. For more information, see [Generate a translation dictionary].

## Benefits of Translations

Localizing Magento storefronts and the Admin panel gives your company global presence for support and sales.

*	Magento supports two types of language packages:

> 	*	Translated Module and theme packages.

Reviewer 2: No period for this and the subsequent bullet since they're not sentences

		Magento auto-discovers packages included in the `i18n` directory of a module or theme. When installing themes and extensions, consider checking for multiple language versions to download and use.

	*	An entire dictionary in one directory.

> 		This is intended to be used and distributed as a standalone component (similar to modules and themes).

Reviewer: This line bugs me a little. It's so...legal-ish. Is it introducing the list?

Author: Tweaked!

* The phrases for translations are enabled in the [Phrase] class.

### More information

See one of the following sections for more information:
* [Theme dictionaries](#m2devgde-xlate-themes)
* [Manually translate words and phrases](#m2devgde-xlate-translating)
* [Translation dictionaries](#m2devgde-xlate-dictionaries)
* [Language packages](#m2devgde-xlate-languagepack)

## Theme dictionaries {#m2devgde-xlate-themes}

You might need to add a dictionary for the default language (en_US) in the following cases:

- To replace or customize strings in the [parent theme]. For example, use "Compare" instead of "Add to Compare".
- To prepare your {% glossarytooltip d2093e4a-2b71-48a3-99b7-b32af7158019 %}theme{% endglossarytooltip %} for localization. More merchants may use your theme if it supports localization.

For an example of creating a dictionary for a theme for both cases, see [Create a translation dictionary for a theme].

## Manually translate words and phrases {#m2devgde-xlate-translating}

Translating the names, titles and phrases used in Magento involves the following steps:

> 1.	Generate a dictionary of your instance using the [translation dictionary tool]. Fully review [Translation dictionaries and packages] to best understand all steps, commands, and best practices for generating the dictionary.

Reviewer 3: I would replace 'best understand' with just 'understand'

*	The Magento code base has the necessary translation dictionaries for a language
*	This language is configured by the store administrator to be used in specified scope (that is, storefront)

The Magento application automatically assembles translation dictionaries located in the modules' `i18n` directory into a dictionary per language. For example, Brazilian Portuguese (`pt_BR`) translation dictionaries might be located in {% glossarytooltip c1e4242b-1f1a-44c3-9d72-1d5b1435e142 %}module{% endglossarytooltip %} and theme directories similar to the following:

* `<Magento_Checkout_module_dir>/i18n/pt_BR.csv`
* `<Magento_Checkout_module_dir>/<theme>/i18n/pt_BR.csv`

{:.bs-callout .bs-callout-info}

> `<Magento_Checkout_module_dir>` stands for the `Magento_Checkout` module directory. The location of this directory depends on the way Magento was installed. See [Conventional notations for paths to modules and themes] for details.

Reviewer 3: is it "on the way Magento was installed" or "where Magento was installed"?

	[Sample language.xml](https://devdocs.magento.com/guides/v2.2/config-guide/cli/config-cli-subcommands-i18n.html#config-cli-subcommands-xlate-example2)

## Community Engineering Translations project {#translations-project}

> Community Engineering accepts translation contributions through the [Magento CrowdIn project] to localize all UI strings. Various meet ups and contribution days include developers contributing translations.

Reviewer 3: meet-ups

1. Create a [CrowdIn account] and join the [Magento CrowdIn project].
1. Browse and select a language. A percentage displays tracking translation progress overall and per section of the Magento Admin.
1. Expand and locate an area of UI text strings to translate. Strings marked red need a translation, and marked green have existing translations.

> 1. Enter or review translations for strings. For more information, see the [CrowdIn knowledge base].

Reviewer 2: ```suggestion
 1. Enter or review translations for strings as described in the [CrowdIn knowledge base].
```
